### PR TITLE
26july countinuse auto slide feat

### PR DESCRIPTION
**Pull request by yakovhaiilo**
What - added the new feat continuousAutoSlide.

this feature allows the option of continuous smooth auto-scrolling.

What I did was integrate the code of the new feature (which gives another option for automatic scrolling) under the same mechanism that already works for interval scrolling,

* now the auto slide is separated into two flows - 'interval scrolling' and "continuous scrolling"

I would love to get constructive comments on everything including names I used etc ...

### CODE REVIEW

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js [hunk 1]
```diff
@@ -386,7 +386,8 @@ export class GalleryContainer extends React.Component {
     fixedScroll,
     isManual,
     durationInMS = 0,
-    scrollMarginCorrection
+    scrollMarginCorrection,
+    isContinuesScrolling = false
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/galleryContainer.js]: continuous

File: packages/lib/test/defaultStyles.spec.js [hunk 2]
```diff
@@ -105,5 +105,7 @@ function expectedStyles() {
     slideTransition: 'cubic-bezier(0.46,0.1,0.25,1)',
     useMaxDimensions: false,
     enableVideoPlaceholder: true,
+    // autoSlideshowType: 'interval',
```
Comment (yhattav) [inline packages/lib/test/defaultStyles.spec.js]: make sure this goes in the default and that the test is green

File: packages/gallery/src/components/gallery/proGallery/galleryContainer.js (hunk 1 above)
Comment (yhattav) [reply packages/gallery/src/components/gallery/proGallery/galleryContainer.js]: ,

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 3]
```diff
@@ -425,8 +447,25 @@ class SlideshowView extends GalleryComponent {
...
       (isGalleryInViewport(this.props.container) || isPreviewMode())
     ) {
-      const direction = this.props.styleParams.isRTL ? -1 : 1;
-      this._next({ direction, isAutoTrigger: true, scrollDuration: 800 });
+      const { styleParams } = this.props;
+      const direction = styleParams.isRTL ? -1 : 1;
+      let isContinuesScrolling = false;
+      let scrollDuration = 800;
+
+      if (
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js:452]: if(cont) next....
else ("interval") next...

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 4]
```diff
@@ -425,8 +447,25 @@ class SlideshowView extends GalleryComponent {
...
+        styleParams.autoSlideshowType ===
+        GALLERY_CONSTS.autoSlideshowTypes.CONTINUES
+      ) {
+        isContinuesScrolling = true;
+        scrollDuration = styleParams.autoSlideshowContinuesSpeed;
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: !==
No speed needed in the function's arguments

File: packages/gallery/src/components/helpers/scrollHelper.js [hunk 5]
```diff
@@ -231,13 +235,18 @@ function horizontalCssScrollTo({
   duration,
   isRTL,
   slideTransition,
+  isContinuesScrolling,
 }) {
-  const change = to - from;
+  let change = to - from;
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/scrollHelper.js:232]: scrollingWidth

File: packages/gallery/src/components/helpers/scrollHelper.js [hunk 6]
```diff
@@ -231,13 +235,18 @@ function horizontalCssScrollTo({
...
 
   if (change === 0) {
     return new Promise((resolve) => resolve(to));
   }
 
+  if (isContinuesScrolling) {
+    duration = Math.abs(change) * duration;
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/scrollHelper.js]: change to speed
wrong calculation

**Review by yhattav — CHANGES_REQUESTED** [review]

File: packages/lib/src/common/coreStyles.js [hunk 7]
```diff
@@ -56,6 +56,8 @@ const coreStyles = {
   scrollDirection: 0,
   slideAnimation: GALLERY_CONSTS.slideAnimations.SCROLL,
   jsonStyleParams: '',
+  autoSlideshowType: GALLERY_CONSTS.autoSlideshowTypes.INTERVAL,
+  autoSlideshowContinuousSpeed: 2,
```
Comment (yhattav) [inline packages/lib/src/common/coreStyles.js]: create a relevant number

File: packages/lib/test/defaultStyles.spec.js [hunk 8]
```diff
@@ -105,5 +105,7 @@ function expectedStyles() {
     slideTransition: 'cubic-bezier(0.46,0.1,0.25,1)',
     useMaxDimensions: false,
     enableVideoPlaceholder: true,
+    autoSlideshowType: 'interval',
+    autoSlideshowContinuousSpeed: 2,
```
Comment (yhattav) [inline packages/lib/test/defaultStyles.spec.js]: relevant default

**Comment by nirnaor** [comment]
@yakovhaiilo  Isn't there a way to keep the scroll logic only inside `slideShowView`? and not in `galleryContainer`?

**Comment by yakovhaiilo** [comment]
> @yakovhaiilo Isn't there a way to keep the scroll logic only inside `slideShowView`? and not in `galleryContainer`?

@nirnaor I see it is also called inside  ״galleryContainer״

**Pull request merged**